test/Elastic.Apm.Tests/HttpDiagnosticListenerTests.cs

```diff
@@ -482,6 +495,8 @@ public async Task HttpCallAsNestedSpan()
 					httpCallSpan.Context.Http.Url.Should().Be($"{localServer.Uri}?i={i}");
 					httpCallSpan.Context.Http.StatusCode.Should().Be(200);
 					httpCallSpan.Context.Http.Method.Should().Be(HttpMethod.Get.Method);
+					httpCallSpan.Context.Destination.Address.Should().Be(new Uri(localServer.Uri).Host);
```

Comment:
Rider warns that `localServer` could be disposed at this point, which is because you access it in an `Action` you pass to a method. Could we assign `localServer.Uri` to a variable before the `numberOfHttpCalls.Repeat` call and pass that variable as parameter?

Reply:
I don't follow - `numberOfHttpCalls.Repeat` doesn't store the closure for later use and it is inside `using (var localServer = new LocalServer())` so I think we can just disable this Rider's warning for this piece of code.

Reply:
BTW we already disable this warning for the same test a few lines above - https://github.com/elastic/apm-agent-dotnet/blob/master/test/Elastic.Apm.Tests/HttpDiagnosticListenerTests.cs#L481-L482

Reply:
> I don't follow - numberOfHttpCalls.Repeat doesn't store the closure for later use and it is inside using (var localServer = new LocalServer()) 

That is true, that is why the code works. But I assume Rider does not detect the fact that the closure runs immediately. Theoretically it could run later. With this, we have to make sure the `Repeat` immediately calls `action()` and we don't replace `Repeat` with anything that runs the closure later. We treat it as a synchronous method call, which it isn't.

I think the warning is valid and I'd rather store `localServer.Uri` in a variable and pass that into the closure - that would always work, no matter when it runs. I think that'd be a cleaner and safer way to get rid of the warning. 

> BTW we already disable this warning for the same test a few lines above -

Yeah, bad, we shouldn't :)

But this is a nit, I'm fine with whatever version.

Reply:
> We treat it as a synchronous method call, which it isn't.

I'm not sure what you mean... The issue is that Rider analysis is not smart enough to understand that this particular closure is not stored for later use. I assume that it's not very easy to make this analysis smart enough because for example in another JetBrains product - Kotlin where passing lambdas around is very big part of the language plus Kotlin has `readonly` for local variables and non-nullable references so compiler has to be able to determine when passed lambda is called and how many times so Kotlin has [contracts](https://pspdfkit.com/blog/2018/kotlin-contracts/) such as `callsInPlace(..., InvocationKind.EXACTLY_ONCE)` to allow developers control its analysis mechanism.

I prefer to disable the warning because storing `localServer.Uri` gives the wrong impression that Rider is right and that the closure can be called later.

Reply:
Ok, I give up - it's not worth arguing on this.